### ISSUE TITLE
Loading pandas categorical breaks when a name contains square bracket character "]"

### ISSUE THREAD

**Issue by pmjpawelec** [jpmml/jpmml-lightgbm#24]
The code doesn't differentiate between syntax "]" and name "]" and throws:
```
Exception in thread "main" java.lang.IllegalArgumentException: ...
at org.jpmml.lightgbm.GBDT.loadPandasCategorical(GBDT.java:460)
...
```

presumably in the second iteration of the `while` loop.

**Comment by 77QingLiu** [comment]
+1

**Comment by vruusmann** [comment]
Blocked by https://github.com/microsoft/LightGBM/issues/1201, which has been superseded by https://github.com/microsoft/LightGBM/issues/960

TLDR: I'm not going to develop a full-blown JavaCC grammar/parser for dealing with complex feature names, because the LightGBM team is about to change this part of LightGBM model files in a major way.

But feel free to "pressurize" the LightGBM team to work on https://github.com/microsoft/LightGBM/issues/960 by upvoting it.

**Comment by pmjpawelec** [comment]
@vruusmann this is understandable. I would add the information to the error message in line 460 though - asking to drop the "]" from names. It might save a lot of time for some people.
